File: Baekjoon/정렬/7795.py
# https://www.acmicpc.net/problem/7795

import sys
import heapq
input = sys.stdin.readline

t = int(input())

for _ in range(t):
    n, m = map(int, input().split())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    
    a.sort()
    b.sort()
    
    result = 0
    cnt = 0
    
    for i in range(n):
        while True:
            if cnt == m or a[i] <= b[cnt]:
                result += cnt
                break
            else:
                cnt += 1

    print(result)

File: Baekjoon/정렬/7795.cpp
// https://www.acmicpc.net/problem/7795

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        vector<int> a, b;

        cin >> n >> m;
        a.resize(n);
        b.resize(m);
        for (int i = 0; i < n; i++) {
            cin >> a[i];
        }
        for (int i = 0; i < m; i++) {
            cin >> b[i];
        }

        sort(a.begin(), a.end());
        sort(b.begin(), b.end());

        int ans = 0;
        for (int i = 0; i < n; i++) {
            int left = 0, right = m;
            while (left < right) {
                int mid = (left + right) / 2;
                if (a[i] > b[mid]) {
                    left = mid + 1;
                } else {
                    right = mid;
                }
            }
            ans += left;
        }

        cout << ans << '\n';
    }

    return 0;
}